sdk/vision/azure-ai-vision-imageanalysis/README.md

```diff
@@ -0,0 +1,338 @@
+
+* [Generate an image caption for an image file](#generate-an-image-caption-for-an-image-file)
+* [Generate an image caption for an image URL](#generate-an-image-caption-for-an-image-url)
+* [Extract text (OCR) from an image file](#extract-text-from-an-image-file)
+* [Extract text (OCR) from an image URL](#extract-text-from-an-image-url)
+
+These snippets use the synchronous `client` from [Create and authenticate the client](#create-and-authenticate-the-client).
+
+See the [Samples](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/vision/azure-ai-vision-imageanalysis/samples) folder for fully working samples for all visual features, including asynchronous clients.
+
+### Generate an image caption for an image file
+
+This example demonstrates how to generate a one-sentence caption for the image file `sample.jpg` using the `ImageAnalysisClient`. The synchronous (blocking) `analyze` method call returns an `ImageAnalysisResult` object with a `caption` property of type `CaptionResult`. It contains the generated caption and its confidence score in the range [0, 1]. By default the caption may contain gender terms such as "man", "woman", or "boy", "girl". You have the option to request gender-neutral terms such as "person" or "child" by setting `gender_neutral_caption = True` when calling `analyze`.
+
+Notes:
+* Caption is only available in some Azure regions. See [Prerequisites](#prerequisites).
+* Caption is only supported in English at the moment.
+
+<!-- SNIPPET:sample_caption_image_file.caption -->
+
+```python
+# Load image to analyze into a 'bytes' object
+with open("sample.jpg", 'rb') as f:
+    image_buffer = bytes(f.read())
+
+# Get a caption for the image. This will be a synchronously (blocking) call.
+result = client.analyze(
+    image_content = image_buffer,
+    visual_features = [ VisualFeatures.CAPTION ],
+    gender_neutral_caption = True) # Optional (default is False)
+
+client.close()
+
+# Print caption results to the console
+print(f"Image analysis results:")
+print(f" Caption:")
+if result.caption is not None:
+    print(f"   '{result.caption.text}', Confidence {result.caption.confidence:.4f}")
+```
+
+<!-- END SNIPPET -->
+
+To generate captions for additional images, simply call `analyze` multiple times. You can use the same `ImageAnalysisClient` do to multiple analysis calls.
+
+### Generate an image caption for an image URL
+
+This example is similar to the above, expect it calls the `analyze` method and provides a [publicly accessible image URL](https://aka.ms/azai/vision/image-analysis-sample.jpg) instead of a file name.
+
+<!-- SNIPPET:sample_caption_image_url.caption -->
+
+```python
+# Get a caption for the image. This will be a synchronously (blocking) call.
+result = client.analyze(
+    image_content = "https://aka.ms/azai/vision/image-analysis-sample.jpg",
+    visual_features = [ VisualFeatures.CAPTION ],
+    gender_neutral_caption = True) # Optional (default is False)
```

Comment:
nit: usually the `)` shall be in new line. You can use `black XX -l 120` to format the code.

Reply:
Thank you @msyyc, I made your suggested changes to the sample code to have the closing bracket in a new line. I'm surprised this was not forced by one of the style checking tools in the pipeline. Perhaps it is optional.